db/migrate/20201105150627_add_messageable_to_progress_messages.rb

```diff
@@ -0,0 +1,25 @@
+class AddMessageableToProgressMessages < ActiveRecord::Migration[5.2]
+  def up
+    add_reference :progress_messages, :messageable, :polymorphic => true
+
+    ProgressMessage.all.each do |message|
+      message.update(
+        :messageable_type => "OrderItem",
+        :messageable_id   => message.order_item_id
+      )
+    end
+
+    # Leave for future to remove
+    # remove_column :progress_messages, :order_item_id, :string
+  end
+
+  def down
+    # add_column :progress_messages, :order_item_id, :string
+
+    ProgressMessage.all.each do |message|
+      message.update!(:order_item_id => message.messageable_id)
+    end
```

Comment:
This...isn't guaranteed to work, right? Because some of the `messageable_id`s will not be `order_item_id`s, but `order_id`s. Then we would either have messages saying they're related to order items that don't exist, or have messages that are related to order items that exist but don't have relevance to them because they're meant to be referencing the order.

I don't think we actually anticipate ever *using* this down migration, but the logic here I think doesn't work. I guess if the message type is "Order" then we should simply delete the progress message?

@bzwei What do you think?

Reply:
yes. we don't expect the rollback will really happen. To make the logic correct we can add a if condition for the case when `message.messageable_type` is `OrderItem`

Reply:
Added logic to handle `Order` and `OrderItem` differently.